inventory/multi_inventory.py

```diff
@@ -136,22 +136,33 @@ def get_provider_tags(self, provider, env=None):
         else:
             cmds.append('--list')
 
-        cmds.append('--refresh-cache')
+        if 'aws' in provider.lower():
+            cmds.append('--refresh-cache')
 
         return subprocess.Popen(cmds, stderr=subprocess.PIPE, \
                                 stdout=subprocess.PIPE, env=env)
 
     @staticmethod
     def generate_config(config_data):
-        """Generate the ec2.ini file in as a secure temp file.
-           Once generated, pass it to the ec2.py as an environment variable.
+        """Generate the provider.ini file in as a secure temp file.
+           Once generated, pass it to the provider as an environment variable.
         """
-        fildes, tmp_file_path = tempfile.mkstemp(prefix='multi_ec2.ini.')
+        prefix = 'multi_inventory.ini.'
+        fildes, tmp_file_path = tempfile.mkstemp(prefix=prefix)
         for section, values in config_data.items():
             os.write(fildes, "[%s]\n" % section)
             for option, value  in values.items():
-                os.write(fildes, "%s = %s\n" % (option, value))
+                # gce.ini files require a path to the private key file.
+                if option in ['gce_service_account_pem_file']:
```

Comment:
I already talked to kwoodson, but this is for posterity. I prefer not having gce_service_account_pem_file hardcoded.